src/_posts/platform/app/2000-01-01-procfile.md

```diff
@@ -9,48 +9,150 @@ Procfile support is one of our
+### Special Process Types
+
+By default, Scalingo uses three process types: `web`, `tcp` and `postdeploy`.
+**These process types have specific use cases** and the platform will only use
+them for what they are designed.
+
+#### The `web` Process Type
+
+The `web` process type defines how to start and run the Managed Runtime
+handling HTTP requests.
+
+It's the only process type that can receive external HTTP requests from
+Scalingo's routers. Consequently, if your application has a web server, it MUST
+have a `web` process type binded on `$PORT` to be publicly reachable.
+
+The `web` process type is often the only one required.
+
+{% note %}
+Almost all our buildpacks define a default `web` process type so you don't have
+to care about it. If you don't need it, you can scale it to 0 to avoid unwanted
+costs.
+{% endnote %}
+
+#### The `tcp` Process Type
+
+The `tcp` process type defines how to start the
+[TCP Gateway addon]({% post_url addons/tcp-gateway/2000-01-01-start %}).
+
+#### The `postdeploy` Process Type
+
+The `postdeploy` process type defines how to start the
+[postdeploy hook]({% post_url platform/app/2000-01-01-postdeploy-hook %}).
+
+
+## Working With the Procfile
+
+### Procfile Naming Convention
+
+The Procfile is a plain text file that MUST be named `Procfile`. Even if it
+uses the YAML syntax, there is **no extension**: no `.txt`, no `.yaml`, no
+`.yml`. **Only `Procfile`**.
 
-The standard syntax is:
+The file MUST live in the application's root directory.
+
+### Defining a Process Type
+
+Each process type of the application is defined on its own, individual line,
+with the following format:
 
 ```yaml
-<container type>: <command>
+<process_type>: <command>
 ```
-where:
 
-- `<container type>` should contain letters and digits only.
-- `<command>` can be any Bash command. It can include environment variables.
+- `<process_type>` designates the name of the process type. It can only contain
+  alphanumerics, dashes and underscores (`[A-Za-z0-9_-]+`). Except this rule,
+  you are free to name your process type how you want: `heavy_process`,
+  `synchronizer`, `scheduler`, `console`, `t-rex`... These are all examples of
+  valid process type names.
+- `<command>` designates the command to run when a Managed Runtime of the
+  specified type is booted up. It can be any Bash command. Using environment
+  variables in the command is allowed, and even encouraged.
 
 {% note %}
-`container type` are unique in the `Procfile`. So, you cannot have, for example, two `web` `container type`.
-If you have, only the second one will be executed, as if the first did not exist.
-If you want to run a command, you may want to look at [Post deploy](https://doc.scalingo.com/platform/app/postdeploy-hook).
+`<process_type>` are unique in the `Procfile`. You cannot have, for example,
+two `web` process types. If you have, the platform will only consider the
+second one. The first one will be ignored, as if it did not exist.
```

Comment:
Not sure that we want to document this behaviour as it might change according to our parser and we might not want to maintain backward compatibility. Maybe saying that the behavior in that case is undefined would be better WDYT ?

Reply:
In fact, I took back what's in the current doc for this one :)

But I agree with you and made the change in https://github.com/Scalingo/documentation/pull/2020/commits/64da5f624d74d705bb2f074587d77ad065c7211c.